packages/block-editor/src/components/warning/index.js

```diff
@@ -14,14 +14,22 @@ import { moreVertical } from '@wordpress/icons';
 function Warning( { className, actions, children, secondaryActions } ) {
 	return (
 		<div style={ { display: 'contents', all: 'initial' } }>
-			<div className={ clsx( className, 'block-editor-warning' ) }>
+			<div
+				className={ clsx( className, 'block-editor-warning' ) }
+				role="alert"
+				aria-live="assertive"
```

Comment:
role alert inherently works as an assertive live region so that `aria-live="assertive"` isn't necessary. `aria-live` doesn;t work well for elements that appear in the DOM anyways.